libs/auth/src/artifact_authentication.ts

```diff
@@ -255,10 +345,13 @@ export async function authenticateArtifactUsingSignatureFile(
       artifact,
       artifactSignatureBundle
     );
+    await performArtifactSpecificValidation(artifact);
```

Comment:
nit: At first glance, it would make more sense to me for "artifact specific validation" to occur outside the main `artifact_authentication` paths, rather than making this code manage the consumers' exceptions. Is there a particular reason to put it here?

Reply:
Ah yeah so the one thing that this function will currently do is recompute the cast vote record root hash to verify that it matches the one in the metadata file. That operation isn't consumer-specific and is in fact a core part of CVR authentication. Without that check, we'd just be verifying that the metadata file hasn't been tampered with. We'd miss any tampering with the many other files in the export.

This comment is making me realizing that this function could use a better name, maybe `performArtifactSpecificAuthenticationChecks`

Reply:
https://github.com/votingworks/vxsuite/pull/3889/commits/39623324179c75d8d1187049e1f328aa88ef9ee0

Reply:
I think the name is fine, I understood the purpose from it. I just was unsure if this was the right place for the hash-checking, both because it applies to only one type of artifact and because right now the import vs. export logic seems asymmetrical. The authenticator is responsible for both creating and checking the signature. And it's responsible for checking the hash, but _not_ creating it, as it needs the hash already created to create the signature file.

I don't see it as problem at all, just observing the patterns as they emerge.

Reply:
> the import vs. export logic seems asymmetrical

This is a really good observation! I found a way to place the hash creation logic in my next PR in `libs/auth`, even though it ultimately has to be the app (VxScan/VxCentralScan) backend that creates the hashes table. So while the asymmetry will still exist, the majority of the hashing logic for export and import will be colocated.

In terms of my hesitance to move the checking logic out to consumers (which I know you're not suggesting), I just want to make it as hard as possible to forget that check. I want the mental model to be "I've called authenticateArtifact, so I'm good to go".